cmd/cluster/core/destroy.go

```diff
@@ -129,17 +131,37 @@ func DestroyCluster(ctx context.Context, hostedCluster *hyperv1.HostedCluster, o
 				o.Log.Error(err, "Failed to get hosted cluster", "namespace", o.Namespace, "name", o.Name)
 				return false, nil
 			}
-			done := len(hostedCluster.Finalizers) == 1 && hostedCluster.Finalizers[0] == destroyFinalizer
+			done := len(hostedCluster.Finalizers) == 1 && hostedCluster.Finalizers[0] == DestroyClusterFinalizer
 			return done, nil
 		}, clusterDeleteCtx.Done())
 		if err != nil {
 			return fmt.Errorf("hostedcluster wasn't finalized, aborting delete: %w", err)
 		}
 	}
 
-	// Destroy additional resources which are specific to the current platform
-	if err := destroyPlatformSpecifics(ctx, o); err != nil {
-		return err
+	// Destroy additional resources which are specific to the current platform (Skip and just remove finalizer with HCI)
+	hci := hyperv1.HostedClusterInfrastructure{
+		ObjectMeta: corev1.ObjectMeta{
+			Name:      o.InfraID,
+			Namespace: o.Namespace,
+		},
+	}
+	hciFound := false
+	if err := c.Get(ctx, client.ObjectKeyFromObject(&hci), &hci); err == nil {
+		hciFound = true
+		//Remove finalizer on HCI
+		origHCI := hci.DeepCopy()
+		controllerutil.RemoveFinalizer(&hci, DestroyClusterFinalizer)
+		if err := c.Patch(ctx, &hci, client.MergeFrom(origHCI)); err != nil {
+			return fmt.Errorf("failed to remove finalizer: %w", err)
+		}
+		o.Log.Info(fmt.Sprintf("HostedClusterInfrastructure resource detected with namespace/name %s/%s",
+			o.Namespace, o.Name))
+	} else {
+		o.Log.Info(fmt.Sprintf("No HostedClusterInfrastructure resource detected with namespace/name %s/%s\n", o.Namespace, o.Name))
```

Comment:
So the destroy will be stopped, if the HCI command fails, the CLI will stop because the aws-creds is missing. I could add an isNotFound check for the legacy path, and for all other errors, log it and exits.

Reply:
Applied the suggestion to both core/destroy.go and core/create.go (now that it doesn't have --use-hci)